File: VBNetCore.Mvvm/Abstractions/IPopupManager.vb
Imports VBNetCore.Mvvm.Controls
Imports VBNetCore.Mvvm.ViewModels

Namespace VBNetCore.Mvvm.Abstractions
    Public Interface IPopupManager
        Function ShowPopup(Of T As PopupViewModel)(ByVal popupViewModel As T, ByVal owner As IPopupable) As T
        Function ShowPopupMessage(ByVal message As String, ByVal caption As String, ByVal owner As IPopupable, ByVal Optional popupButton As PopupButton = PopupButton.OK, ByVal Optional popupImage As PopupImage = PopupImage.None) As PopupResult
    End Interface
End Namespace

File: VBNetCore.Mvvm/Abstractions/IPopupManager.cs
using NetCore.Mvvm.Controls;
using NetCore.Mvvm.ViewModels;

namespace NetCore.Mvvm.Abstractions
{
    public interface IPopupManager
    {
        T? ShowPopup<T>(T popupViewModel, IPopupable owner) where T : PopupViewModel;
        PopupResult ShowPopupMessage(string message, string caption, IPopupable owner, PopupButton popupButton = PopupButton.OK, PopupImage popupImage = PopupImage.None);
    }
}
